scripts/build_qat_envoy.sh

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -e
+
+if [ -f /etc/os-release ]; then
+	source /etc/os-release
+else
+	source /usr/lib/os-release
+fi
+
+readonly YQ_BIN="$(realpath ${PWD}/yq)"
+readonly BAZEL_BIN="$(realpath ${PWD}/bazel.sh)"
+readonly SCRIPTS_DIR="$(realpath "$(dirname "$0")")"
+readonly VERSIONS_FILE="${SCRIPTS_DIR}/../versions.yaml"
+readonly OPENSSL_DIR="${SCRIPTS_DIR}/../openssl"
+readonly QAT_ENGINE_DIR="${SCRIPTS_DIR}/../QAT_Engine"
+readonly ENVOY_DIR="${SCRIPTS_DIR}/../envoy"
+readonly QAT_LIB_DIR="${SCRIPTS_DIR}/../QAT_Lib"
+
+source "${SCRIPTS_DIR}/lib.sh"
+
+install_yq() {
+	info "Download and install yq"
+	local yq_arch=""
+	case $(uname -m) in
+		x86_64)
+			yq_arch=amd64
+		;;
+
+		*)
+			die "Unsupported architecture: $(uname -m)"
+	esac
+
+	curl -L https://github.com/mikefarah/yq/releases/download/2.4.0/yq_linux_${yq_arch} -o ${YQ_BIN}
+	chmod +x ${YQ_BIN}
+}
+
+install_bazel() {
+	info "Download and install bazel"
+	local bazel_url="$(${YQ_BIN} r "${VERSIONS_FILE}" "tools.bazel.url")"
+	local bazel_version="$(${YQ_BIN} r "${VERSIONS_FILE}" "tools.bazel.version")"
+	curl -L "${bazel_url}/releases/download/${bazel_version}/bazel-${bazel_version}-installer-linux-x86_64.sh" -o "${BAZEL_BIN}"
+	chmod +x "${BAZEL_BIN}"
+}
+
+setup() {
+	case $ID in
+		clear-linux*)
+			export HOME=/root
+			info "Clear Linux OS detected"
+			rm -rf /run/lock/clrtrust.lock
+			clrtrust generate
+			swupd update
+			swupd bundle-add os-core-dev
+			;;
+		debian|ubuntu)
+			info "Debian/Ubuntu OS detected"
+			# https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=863199
+			 mkdir -p /usr/share/man/man1
+			 apt-get -o Acquire::Check-Valid-Until=false update
+			 apt-get -y install git libtool \
+					 cmake clang-format-7 automake ninja-build curl \
+					 git build-essential wget libudev-dev libssl-dev \
+					 openssl pkg-config autoconf autogen libtool \
+					 libssl-dev pkg-config zip g++ zlib1g-dev unzip \
+					 python curl gnupg2
+			;;
+	esac
+
+	install_yq
+
+	install_bazel
+	"${BAZEL_BIN}" --user
+
+}
+
+build_install_qat_library() {
+	local qat_url="$(${YQ_BIN} r "${VERSIONS_FILE}" "libraries.qat.url")"
+	local qat_version="$(${YQ_BIN} r "${VERSIONS_FILE}" "libraries.qat.version")"
+
+	mkdir -p "${QAT_LIB_DIR}"
+	pushd "${QAT_LIB_DIR}"
+
+	curl -L "${qat_url}/qat${qat_version}.tar.gz" -o qat.tz
+	tar -xf qat.tz
+
+	sed -i -e 's/cmn_ko$//' quickassist/Makefile
+	export ICP_ROOT="${QAT_LIB_DIR}" ICP_BUILD_OUTPUT="${QAT_LIB_DIR}/build" \
+		   ICP_ENV_DIR="${QAT_LIB_DIR}/quickassist/build_system/build_files/env_files" \
+		   ICP_BUILDSYSTEM_PATH="${QAT_LIB_DIR}/quickassist/build_system" KERNEL_SOURCE_ROOT=/tmp
+
+	case $ID in
+		clear-linux*)
+			./configure --enable-qat-uio
```

Comment:
Note that required command line options for `libqat` compilation differ from version to version.